dns/route53/dns_test.go

```diff
@@ -0,0 +1,115 @@
+package route53
+
+import (
+	"testing"
+	"time"
+
+	"fmt"
+	"net"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+)
+
+type mockR53 struct {
+	recordSet *route53.ResourceRecordSet
+	zoneID string
+}
+
+func (r53 *mockR53) ChangeResourceRecordSets(input *route53.ChangeResourceRecordSetsInput) (*route53.ChangeResourceRecordSetsOutput, error) {
+	// NOTE: if the ChangeBatch would contain more than just one Change, they would be ignored
+	change := input.ChangeBatch.Changes[0]
+
+	switch *change.Action {
+	case route53.ChangeActionCreate, route53.ChangeActionUpsert:
+		r53.recordSet = change.ResourceRecordSet
+
+	case route53.ChangeActionDelete:
+		r53.recordSet = nil
+
+	default:
+		panic(fmt.Sprintf("Invalid action: %s", *change.Action))
```

Comment:
We can `return nil, fmt.Errorf(…)` here.

Reply:
I consider this a developer mistake, hence the `panic` - making the author aware of this asap.